### PR TITLE
[docs][large] Replace mdjs scripts with MDX

### PR DESCRIPTION
**Pull request by nicknovitski**
MDX is the JSX of markdown: it lets us import and use react components in the middle of markdown files. 

# Why

The mdjs steps are a snowflake and I wanted to get rid of them.  This change replaces our "compilation" scripts with webpack and babel.  It also fixes live reloading, at least for me.

In addition, the change makes it much much easier add rich content to the docs.  No more adding special-case regexes in a weird corner of the repository, just write and import components.  

# How

I tried to make as few changes as possible, knowing that they would be enormous anyway.  Unfortunately it looks like github is not detecting file similarity very well...
 
In practice the only user-facing change I had to make was in the standard formatting for indicating the types of parameters in the sdk pages: text like `Array<number>` is now interpreted as opening the html element `<number>`, unless wrapped in backticks.

# Test Plan

You should check out this branch locally, run a dev server, and check everything you think might break.

(I made sure tippy worked.)

### CODE REVIEW

**Comment by nicknovitski** [comment]
The conflicts are from the `Expo` namespace removal,  I'll get to them.

**Comment by nicknovitski** [comment]
Ah, on the files changed tab, you can use the filter files button to hide all the `.md` files.  Again, their only changes were
- Add import and export lines
- Occasional whitespace additions
- Remove unneeded `${...}`
- Change all parameter formatting to backticks
- Change code blocks with `es6` language to `javascript` (prism doesn't have an `es6` definition)

File: docs/components/base/code.js [hunk 1]
```diff
@@ -101,9 +101,8 @@ export class Code extends React.Component {
   }
 
   render() {
-    let code = this.props.children;
-    let { lang } = this.props;
-    let html = code.toString();
+    let lang = this.props.className && this.props.className.slice(9).toLowerCase();
```
Comment (brentvatne) [inline docs/components/base/code.js]: might be worth adding a comment about why this is necessary

File: docs/package.json [hunk 2]
```diff
@@ -3,19 +3,18 @@
   "version": "32.0.0",
   "private": true,
   "scripts": {
-    "dev": "concurrently -k 'yarn run mdjs' 'node server.js'",
-    "build": "yarn run mdjs-prod && cross-env NODE_ENV=production NODE_OPTIONS=--max-old-space-size=8192 next build",
```
Comment (brentvatne) [inline docs/package.json]: intentional to remove NODE_ENV for build and export?

File: docs/package.json [hunk 3]
```diff
@@ -3,19 +3,18 @@
...
-    "export": "yarn run build && cross-env NODE_ENV=production next export",
+    "dev": "node server.js",
```
Comment (brentvatne) [inline docs/package.json]: nice!

**Review by brentvatne — APPROVED** [review]
a couple of small questions but looks good to me!

File: docs/package.json (hunk 2 above)
Comment (nicknovitski) [reply docs/package.json]: Yeah.  [It defaults to `production` for every subcommand except `dev`](https://github.com/zeit/next.js/blob/c6d3484a2112abceebf724fbabb80840ec24e6d1/packages/next/bin/next.ts#L84-L85).

**Pull request merged**